File: template/aspnet/function/Function.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace OpenFaaS
{
    [ApiController]
    [Route("/")]
    public class Function : ControllerBase
    {
        [HttpGet]
        [HttpPost]
        public Task<IActionResult> ExecuteAsync()
        {
            var result = new
            {
                Message = "Hello!"
            };

            return Task.FromResult<IActionResult>( Ok( result ) );
        }
    }
}

File: template/aspnet-fsharp/function/Function.fs
namespace OpenFaaS

open System
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc

[<ApiController>]
[<Route("/")>]
type Function() =
    inherit ControllerBase()

    [<HttpGet>]
    [<HttpPost>]
    member this.ExecuteAsync() =
        let result = {| Message = "Hello" |}
        Task.FromResult( this.Ok( result ) :> IActionResult )
